mono/mini/mini-exceptions.c

```diff
@@ -2037,7 +2057,7 @@ mono_handle_exception_internal (MonoContext *ctx, MonoObject *obj, gboolean resu
 
 			if (unhandled)
 				mono_debugger_agent_handle_exception ((MonoException *)obj, ctx, NULL, NULL);
-			else
+			else if (res != MONO_FIRST_PASS_CALLBACK_TO_NATIVE)
```

Comment:
If we got here, res is `MONO_FIRST_PASS_UNHANDLED` so this check seems redundant.

Reply:
I don't think so - this is all in the `else` branch of `if (res == MONO_FIRST_PASS_UNHANDLED)`, so it could be handled or passed to native.

Reply:
Oh yes, misread the code.